### PR TITLE
chore: update switchToDomain 'data' argument to be an array

### LINKED ISSUE #19708: Multi-domain: Change switchToDomain data argument to only accept an array
### What would you like?

- Change the `data` argument of `switchToDomain`, requiring it to be an array instead of any serializable value.
- Validate that it's an array and throw an error if not
- The argument will still be provided as-is as the first argument to the callback function. Users can use destructuring to spread the values out.

```js
const data = ['janedoe', 's3cr3t']

cy.switchToDomain('foobar.com', data, (data) => {
  const username = data[0]
  const password = data[1]
 // ...
})

// - or -

cy.switchToDomain('foobar.com', data, ([username, password]) => {
 // ...
})
```

### Why is this needed?

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by mschile**
- Closes #19708 

### User facing changelog

n/a

### Additional details

* Updated the `switchToDomain` `data` argument to be an array
  * The array can be of any type including mixed types
* If an array is not passed, an error is thrown notifying the user the `data` argument must be an array

### PR Tasks

- [x] Have tests been added/updated?
- [n/a] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [x] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [n/a] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Review by chrisbreiding — CHANGES_REQUESTED** [review]
Looks good 👍 

Just need to update [the public types](https://github.com/cypress-io/cypress/blob/7ef104e9a8657f49751b422b2978a4fc1e5d2963/cli/types/cypress.d.ts#L1923-L1942).

**Comment by mschile** [comment]
> Just need to update [the public types](https://github.com/cypress-io/cypress/blob/7ef104e9a8657f49751b422b2978a4fc1e5d2963/cli/types/cypress.d.ts#L1923-L1942).

Updated https://github.com/cypress-io/cypress/pull/19774/commits/42139419c7e8e9e85ba87307a47de5559d56fb8b

File: cli/types/cypress.d.ts [hunk 1]
```diff
@@ -1916,11 +1916,12 @@ declare namespace Cypress {
      * Enables running Cypress commands in a secondary domain
      * @see https://on.cypress.io/switchToDomain
      * @example
-     *    cy.switchToDomain('example.com', { key: 'value' } ({ key }) => {
+     *    cy.switchToDomain('example.com', [{ key: 'value' }, 'foo'] ([{ key }, foo]) => {
```
Comment (chrisbreiding) [inline cli/types/cypress.d.ts:1919]:
```suggestion
     *    cy.switchToDomain('example.com', [{ key: 'value' }, 'foo'], ([{ key }, foo]) => {
```

File: cli/types/cypress.d.ts [hunk 2]
```diff
@@ -1916,11 +1916,12 @@ declare namespace Cypress {
...
      *      expect(key).to.equal('value')
+     *      expect(foo).to.equal('foo')
      *    })
      */
-    switchToDomain<T>(domain: string, data: T, fn: (data: T) => void): Chainable<undefined>
+    switchToDomain(domain: string, data: [], fn: (data: []) => void): Chainable<undefined>
```
Comment (chrisbreiding) [inline cli/types/cypress.d.ts:1924]:
```suggestion
    switchToDomain<T>(domain: string, data: T[], fn: (data: T[]) => void): Chainable<undefined>
```

I think we can keep the generic for these like this, right?

File: cli/types/cypress.d.ts [hunk 3]
```diff
@@ -1929,17 +1930,18 @@ declare namespace Cypress {
...
      * @example
-     *    cy.switchToDomain('example.com', done, { key: 'value' } ({ key }) => {
+     *    cy.switchToDomain('example.com', done, [{ key: 'value' }, 'foo'], ([{ key }, foo]) => {
      *      expect(key).to.equal('value')
+     *      expect(foo).to.equal('foo')
      *      done()
      *    })
      */
-    switchToDomain<T>(domain: string, done: Mocha.Done, data: T, fn: (data: T) => void): Chainable<undefined>
+    switchToDomain(domain: string, done: Mocha.Done, data: [], fn: (data: []) => void): Chainable<undefined>
```
Comment (chrisbreiding) [inline cli/types/cypress.d.ts:1944]:
```suggestion
    switchToDomain<T>(domain: string, done: Mocha.Done, data: T[], fn: (data: T[]) => void): Chainable<undefined>
```

**Review by chrisbreiding — CHANGES_REQUESTED** [review]
Also need to update the [types tests](https://github.com/cypress-io/cypress/blob/7ef104e9a8657f49751b422b2978a4fc1e5d2963/cli/types/tests/cypress-tests.ts#L831).

File: cli/types/cypress.d.ts (hunk 2 above)
Comment (mschile) [reply cli/types/cypress.d.ts]: Yeah, you're right, we can keep the generic.

File: cli/types/cypress.d.ts (hunk 1 above)
Comment (mschile) [reply cli/types/cypress.d.ts]: Update [here](9817bf15d5397f7edaedf21be9700be8e565fcc8)

File: cli/types/cypress.d.ts (hunk 2 above)
Comment (mschile) [reply cli/types/cypress.d.ts]: [Updated here](d1d6599ea65312b86ae2ffcd9018178a1bf2c732)

File: cli/types/cypress.d.ts (hunk 3 above)
Comment (mschile) [reply cli/types/cypress.d.ts]: Updated [here](d1d6599ea65312b86ae2ffcd9018178a1bf2c732)

**Comment by mschile** [comment]
> Also need to update the [types tests](https://github.com/cypress-io/cypress/blob/7ef104e9a8657f49751b422b2978a4fc1e5d2963/cli/types/tests/cypress-tests.ts#L831).

Updated [here](d1d6599ea65312b86ae2ffcd9018178a1bf2c732)

**Review by chrisbreiding — APPROVED** [review]

**Review by AtofStryker — APPROVED** [review]

**Review by mjhenkes — APPROVED** [review]

**Pull request merged**